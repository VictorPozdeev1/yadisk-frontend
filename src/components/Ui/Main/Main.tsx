import React, { FC, PropsWithChildren } from 'react';
import { Outlet } from 'react-router';
import { DocumentTable } from '../DocumentTable/DocumentTable';
import Box from '@mui/material/Box';
import { display } from '@mui/system';

export interface MainProps {

}
export const Main: FC<PropsWithChildren<MainProps>> = () => {

  return (
    <main style={{
      display: 'flex',
      flexDirection: 'column'
    }}>
      <DocumentTable />
      <Outlet />
    </main>
  )
}
