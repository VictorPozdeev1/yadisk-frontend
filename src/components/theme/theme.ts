import { createTheme, responsiveFontSizes } from '@mui/material/styles';
import { Theme } from '@mui/system';

// переопределяем инерфейс брейкпоинтов
declare module '@mui/material/styles' {
  interface BreakpointOverrides {
    xs: false; // removes the `xs` breakpoint
    sm: false;
    md: false;
    lg: false;
    xl: false;
    mobile: true; // adds the `mobile` breakpoint
    tablet: true;
    laptop: true;
    desktop: true;
  }
}
const themeColors={
  fontDark:"#181818",
  actionBlue:'#035FA2',
  secondary:'#EDEDED',

}
export const appTheme:Theme = createTheme({
  typography: {
    fontFamily: 'Manrope, Roboto, Arial',

    h1:{
      fontWeight:'bold',
    },
    h2:{//category title
      fontSize:36,
      lineHeight:1.5,
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h3:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h4:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h5:{
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    h6:{//doc table header
      color:themeColors.fontDark,
      fontWeight:'bold',
    },
    body1:{
      color:themeColors.fontDark,
      fontWeight:'600',
    },
  },
  breakpoints:{
    values:{
      mobile:320,
      tablet:805,
      laptop:1200,
      desktop:1523,
    }
  },

  palette: {
    primary: {
      main: themeColors.actionBlue,//action-blue
      light: themeColors.secondary,
      contrastText: themeColors.fontDark,//font-dark
    },
    secondary:{//for table buttons
      main: themeColors.fontDark,//font-dark
    }
  },

});
