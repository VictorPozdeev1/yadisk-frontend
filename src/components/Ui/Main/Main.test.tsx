import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Main } from './Main'
import { ThemeProvider } from "@mui/material";
import { appTheme } from '../../theme/theme';

describe('Main component', () => {
    test('Main render', () => {
        render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <Main />
                </ThemeProvider>
            </BrowserRouter>
        )

        expect(screen.getByRole('main')).toBeInTheDocument();
        expect(screen.getByText('Все документы')).toBeInTheDocument();
    })

    test('Main snapshot', () => {
        const main = render(
            <BrowserRouter>
                <ThemeProvider theme={appTheme}>
                    <Main />
                </ThemeProvider>
            </BrowserRouter>
        )
        expect(main).toMatchSnapshot();
    })
})