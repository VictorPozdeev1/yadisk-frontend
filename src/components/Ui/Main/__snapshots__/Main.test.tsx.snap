// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Main snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1m96vbc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1qc0fec-MuiTypography-root"
          >
            Все документы
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        />
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1m96vbc-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1qc0fec-MuiTypography-root"
        >
          Все документы
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      />
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
